packages/block-editor/src/components/inserter/hooks/use-insertion-point.js

```diff
@@ -108,41 +113,40 @@ function useInsertionPoint( {
 					selectedBlock.clientId,
 					blocks,
 					null,
-					null,
+					shouldFocusBlock ? 0 : undefined,
 					meta
 				);
 			} else {
 				insertBlocks(
 					blocks,
 					destinationIndex,
 					destinationRootClientId,
-					selectBlockOnInsert,
+					shouldFocusBlock,
 					meta
 				);
 			}
-
-			if ( ! selectBlockOnInsert ) {
+			const message = sprintf(
 				// translators: %d: the name of the block that has been added
-				const message = _n(
+				_n(
 					'%d block added.',
 					'%d blocks added.',
-					blocks.length
-				);
-				speak( message );
-			}
+					castArray( blocks ).length
+				),
+				castArray( blocks ).length
+			);
+			speak( message );
```

Comment:
This should fix #28151. Maybe we should create a separate PR for that so this fix doesn't depend on the other changes here?

Reply:
You're right, done in https://github.com/WordPress/gutenberg/pull/28871